tests/ops/qubit/test_non_parametric_ops.py

```diff
@@ -915,3 +915,69 @@ def test_control_wires(op, control_wires):
     """Test ``control_wires`` attribute for non-parametrized operations."""
 
     assert op.control_wires == control_wires
+
+
+all_ops = [
+    qml.Identity(0),
+    qml.Hadamard(0),
+    qml.PauliX(0),
+    qml.PauliY(0),
+    qml.PauliZ(0),
+    qml.S(wires=0),
+    qml.T(wires=0),
+    qml.SX(wires=0),
+    qml.CNOT(wires=(0, 1)),
+    qml.CZ(wires=(0, 1)),
+    qml.CY(wires=(0, 1)),
+    qml.SWAP(wires=(0, 1)),
+    qml.ISWAP(wires=(0, 1)),
+    qml.SISWAP(wires=(0, 1)),
+    qml.SQISW(wires=(0, 1)),
+    qml.CSWAP(wires=(0, 1, 2)),
+    qml.Toffoli(wires=(0, 1, 2)),
+    qml.MultiControlledX(control_wires=(0, 1, 2), wires=(3)),
+    qml.Barrier(0),
+    qml.WireCut(wires=0),
+]
+
+idempotent_ops = [
+    qml.Identity,
+    qml.Hadamard,
+    qml.PauliX,
+    qml.PauliY,
+    qml.PauliZ,
+    qml.CNOT,
+    qml.CZ,
+    qml.CY,
+    qml.SWAP,
+    qml.CSWAP,
+    qml.Toffoli,
+    qml.MultiControlledX,
+    qml.Barrier,
+    qml.WireCut,
+]
+
+
+@pytest.mark.parametrize("op", all_ops)
+@pytest.mark.parametrize("num_adjoint_calls", [1, 2, 3])
```

Comment:
It might be easier to move this parametrization inside the test. This way is fine, but we could also:
* take one adjoint
* validate
* take second adjoint
* validate
* take third adjoint
* validate

instead of taking it multiple times in a row then validating.